### ISSUE TITLE
[Feature request] Add send amount in `USD`

### ISSUE THREAD

**Issue by kolya182** [sparrowwallet/sparrow#147]
As far as we have price in app would be very helpful to send USD amount(when exact etc amount is not the case).
Use cases:
* Send a friend 10$ in BTC to hook him up
* Send Sparrow dev 20$donation, calculating `sats` amount is just another obstacle and friction and it takes time. Will send once implemented.

<img width="627" alt="Screen Shot 2021-06-24 at 9 10 49 PM" src="https://user-images.githubusercontent.com/3277207/123368436-a9de5980-d530-11eb-837d-65794bce3bac.png">

Example from Trust Wallet:
![IMAGE 2021-06-24 21:13:10](https://user-images.githubusercontent.com/3277207/123368622-fd50a780-d530-11eb-9513-0520c00ddb65.jpg)

**Comment by RequestPrivacy** [comment]
Not sure if that's the path forward. Why would you want to measure you bitcoin in depreciating fiat? It's "convenient" bc it aligns with how we measure value today, I get that. But to overcome a broken system you have to change your mind-setting. In your mind send him bitcoin, not fiat. Send him how much he needs to fund a trade in HodlHodl or [Bisq](https://github.com/bisq-network/bisq).

**Comment by kolya182** [comment]
> Not sure if that's the path forward. Why would you want to measure you bitcoin in depreciating fiat? It's "convenient" bc it aligns with how we measure value today, I get that. But to overcome a broken system you have to change your mind-setting. In your mind send him bitcoin, not fiat. Send him how much he needs to fund a trade in HodlHodl or [Bisq](https://github.com/bisq-network/bisq).

Don't be maxi

**Comment by marcocattaneogotam** [comment]
+1

**Comment by marcocattaneogotam** [comment]
I really dont get it.
Any bitcoin wallet allows this action, and it's pretty obvious: you agree with the recipient the transaction amount in fiat, so how can you send the exact amount from sparrow?

**Comment by PrinceOfEgypt** [comment]
I also agree. The amount is shown in fiat when you enter a BTC / sat amount. You should be able to enter the amount in the configured fiat currency as well.

**Comment by RequestPrivacy** [comment]
I really do hope that all you guys begging for being able to send "USD/fiat" amounts get rug pulled by a faulty exchange rate so that you accidentally "send" $5 but with an exchange rate of $1/BTC...goodbye to your 5 BTC you just transferred.

**Comment by PrinceOfEgypt** [comment]
Do you also worry about entering 50,000 BTC instead of 50,000 sats? Or is this just an excuse because you have this maxi purist position? You were the one who started the insane issue to remove all fiat references from the app after all, right?

**Comment by RequestPrivacy** [comment]
No, it's not an excuse, just a wish. 
No, it's not a maxi position as those would argue more in the line of you & all the others (make it easy for newcomers, mass adoption, ngu,  I just want to send my fiat bro...)

Yes, it's a purist view. 
Yes, I started the "insance" discussion to remove all fiat references (how do other people use wallets where no fiat is displayed....magicians?)
Yes, I think the more fiat in a wallet appears the dumber it's users are and/or will become. 
Yes, you will see that as an insult but it's like with every thing: put enough abstractions on top and nobody will bother to still understand the fundamentals. 

I'll give this space 2 more halvings and most "bitcoiners" won't even understand why their wallet sends 2 different btc amounts if they send the exact same USD amount at different times.

**Comment by PrinceOfEgypt** [comment]
> No, it's not an excuse, just a wish. 
> 
> No, it's not a maxi position as those would argue more in the line of you & all the others (make it easy for newcomers, mass adoption, ngu,  I just want to send my fiat bro...)
> 
> 
> 
> Yes, it's a purist view. 

Ok, great. Make it an option - you should be able to hide all fiat values if you don't want to see them

> 
> Yes, I started the "insane" discussion to remove all fiat references (how do other people use wallets where no fiat is displayed....magicians?)
> 

I literally can't think of a single wallet like off of the top of my head. I'm sure they exist, just like there all people out there who don't run a DE on their Linux box. It's the exception, not the rule. 

> Yes, I think the more fiat in a wallet appears the dumber it's users are and/or will become. 
> 
> Yes, you will see that as an insult but it's like with every thing: put enough abstractions on top and nobody will bother to still understand the fundamentals. 
> 

Then why allow public servers? Why allow turning off tor? Why allow "efficiency" transactions? Maybe every sparrow user should run a full node, fulcrum server, and sparrow should only allow coinjoin transactions?

> 
> I'll give this space 2 more halvings and most "bitcoiners" won't even understand why their wallet sends 2 different btc amounts if they send the exact same USD amount at different times.

Sorry, but that's BS. We're the ones running our own nodes and electrum servers, and we're saying right now there's value still in thinking in terms of both BTC and fiat. People understand what an exchange rate is, no one is asking sparrow to protect them from their own stupidity. An advanced wallet should have advanced features but still be accessible to new users.

**Comment by PrinceOfEgypt** [comment]
> Tor can be DDoS'ed so you need an alternative way to connect (e.g. via VPS). "efficient" tx are normal bitcoin transactions why should they be a problem (if not for privacy)? Public servers are also bad for privacy but other than that, why not? 
> 
But that's my point - why allow these compromises (breaking privacy, etc) but be so hung up on showing fiat values?

**Comment by RequestPrivacy** [comment]
Because they are inherent system limitations of bitcoin, networking etc. but FIAT isn't.

**Comment by RequestPrivacy** [comment]
>  Sparrow is offering a compromise that violates the "best practices" for the sake of usability.

You have it completely backwards: it offers different ways to connect to the bitcoin network and offers from standard to privacy enhancing transactions everything. The focus is on _enhancing_. 

As I said, these enhancements come all down to solutions to basic problems on the protocol or networking layer.

YOU on the other hand impose an outside unit of account (fiat) ONTO bitcoin which simply has no relevance in the world where the wallet operates. The wallet doesn't send fiat, the wallet doesn't receive fiat so it makes no sense to try and stuff a $5 "Send" button onto the wallet because it CAN'T SEND your imaginary exact $5.

**Comment by PrinceOfEgypt** [comment]
> You have it completely backwards: it offers different ways to connect to the bitcoin network and offers from standard to privacy enhancing transactions everything. The focus is on _enhancing_.
> 
> As I said, these enhancements come all down to solutions to basic problems on the protocol or networking layer.

OK, great. Right now there is a basic problem - no matter how much you want to deny it, exchange rates are a thing - Bitcoin is still expressed in BTC-USD, or BTC-EUR, etc. No one is forcing you to think in terms of the exchange rate if you don't want to. Go use Samourai if you don't want to see fiat values. you've been making this argument since 2021 - has **anyone** really stopped thinking in terms of the fiat value of BTC since then?

**Comment by RequestPrivacy** [comment]
> Right now there is a basic problem - no matter how much you want to deny it, exchange rates are a thing - Bitcoin is still expressed in BTC-USD, or BTC-EUR, etc.

I don't deny as you proclaim. As you very rightly stated there's a basic problem and that is people pricing Bitcoin and goods in fiat.

I would describe the conflict we have more in terms that people think differently about this problem and the consequences of how it should be handled. You all want to make life easy by removing the pain of having to deal with native Bitcoin by making it as much fiat as possible. I guess for "inclusiveness" you'd be willing to remove every bit that normies could find irritating on Bitcoin just so they're "onboard". I further guess that you think somewhere in the future (on a supposed bitcoin standard) this all magical reverses and people deal in sats.

I think this describes the slow death of Bitcoin as it will never happen. If you dance with the devil, the devil doesn't change, it changes you.

> has anyone really stopped thinking in terms of the fiat value of BTC since then?

That's like regulators pointing at a market fully distorted from their regulations and screaming "look capitalist, your free market doesn't work, we need more regulations".

How on earth should people learn to price things in bitcoin if all you do is put more fiat on top so that they don't have to? You're really blaming me that no one adopts a btc denomination while crying that people need fiat in a wallet? Really?
